elastic.js/index.d.ts

```diff
@@ -6765,7 +6765,7 @@ declare module elasticjs {
     /*
      Allows you to set a specified filter on this request object.
      */
-    filter(filter: Object): Request;
+    filter(filter: Object | Filter): Request;
```

Comment:
@bfsmith 
Thank you for you merge-request.

Looks OK.

And yes, I agree with @DanielRosenwasser that it may be just collapsed to `Filter` as it is already done for facets: `facet(facet: Facet): Request;`